bayesian_testing/experiments/delta_normal.py

```diff
@@ -0,0 +1,298 @@
+            }
+        elif name in self.variant_names and replace:
+            msg = (
+                f"Variant {name} already exists - new data is replacing it. "
+                "If you wish to append instead, use replace=False."
+            )
+            logger.info(msg)
+            self.data[name] = {
+                "totals": totals,
+                "positives": positives,
+                "sum_values": sum_values,
+                "sum_values_2": sum_values_2,
+                "a_prior_beta": a_prior_beta,
+                "b_prior_beta": b_prior_beta,
+                "m_prior": m_prior,
+                "a_prior_ig": a_prior_ig,
+                "b_prior_ig": b_prior_ig,
+                "w_prior": w_prior,
+            }
+        elif name in self.variant_names and not replace:
+            msg = (
+                f"Variant {name} already exists - new data is appended to variant, "
+                "keeping its original prior setup. "
+                "If you wish to replace data instead, use replace=True."
+            )
+            logger.info(msg)
+            self.data[name]["totals"] += totals
+            self.data[name]["positives"] += positives
+            self.data[name]["sum_values"] += sum_values
+            self.data[name]["sum_values_2"] += sum_values_2
+
+    def add_variant_data(
+            self,
+            name: str,
+            data: List[Number],
+            a_prior_beta: Number = 0.5,
+            b_prior_beta: Number = 0.5,
+            m_prior: Number = 1,
+            a_prior_ig: Number = 0,
+            b_prior_ig: Number = 0,
+            w_prior: Number = 0.01,
+            replace: bool = True,
+    ) -> None:
+        """
+        Add variant data to test class using raw Delta-Normal data.
+
+        The goal of default prior setup is to be low information. It should be tuned with caution.
+
+        Parameters
+        ----------
+        name : Variant name.
+        data : List of delta-normal data (e.g. revenues in sessions).
+        a_prior_beta : Prior alpha parameter from Beta distribution for conversion part.
+        b_prior_beta : Prior beta parameter from Beta distribution for conversion part.
+        m_prior : Prior normal mean.
+        a_prior_ig : Prior alpha from inverse gamma dist. for unknown variance.
+            In theory a > 0, but as we always have at least one observation, we can start at 0.
+        b_prior_ig : Prior beta from inverse gamma dist. for unknown variance.
+            In theory b > 0, but as we always have at least one observation, we can start at 0.
+        w_prior : Prior effective sample sizes.
+        replace : Replace data if variant already exists.
+            If set to False, data of existing variant will be appended to existing data.
+        """
+        if len(data) == 0:
+            raise ValueError("Data of added variant needs to have some observations.")
+
+        totals = len(data)
+        positives = sum(x > 0 for x in data)
```

Comment:
this is likely wrong. If data can be negative I this it should be called `non_zeros` and rule should be more like `x != 0`.